mk/spksrc.checksum.mk

```diff
@@ -39,18 +39,13 @@ pre_checksum_target: checksum_msg
 
 # validate file integrity with the provided digests.
 checksum_target: $(PRE_CHECKSUM_TARGET)
-	@if [ -f $(DIGESTS_FILE) ] ; \
-	then \
-	  cat $(DIGESTS_FILE) | (cd $(DISTRIB_DIR) && while read file type value ; \
+	@if [ -f $(DIGESTS_FILE) ] ; then \
+	  cat $(DIGESTS_FILE) | grep $(LOCAL_FILE) | ( \
+	  count=0 ; \
```

Comment:
The `count` variable could be discarded entirely if there was a leading if statement to check if `grep` is not null.
As such there could be preceding this a simple check if/else such as:
```
if [ $(cat $(DIGESTS_FILE) | grep -c $(LOCAL_FILE) || true) -eq 0 ]; then
   $(MSG) "  Downloaded file $(LOCAL_FILE) is not in digests file"
else
   ...
```

Reply:
@th0ma7 thanks for the tip, I first wanted to cound for the 3 checksum types, but the regular checks do not count either.

Reply:
Finally I added the verification for at least 3 checksums.

Reply:
Then again, the `count` does not provide that much as it does not validate each `md5`|`sha` types.
You could do instead, right from the get go:
```
# File missing in digests
if [ $(cat $(DIGESTS_FILE) | grep -c $(LOCAL_FILE) || true) -eq 0 ]; then
   $(MSG) "  Downloaded file $(LOCAL_FILE) is not in digests file"
   rm $(DOWNLOAD_COOKIE)
# Missing sha entries in digests
elif [ $(cat $(DIGESTS_FILE) | sed '/MD5/{/SHA1/{/SHA256/p;};}' | grep -c $(LOCAL_FILE) || true) -lt 3 ]; then
   $(MSG) "  Downloaded file $(LOCAL_FILE) has only $$count/3 checksum entries. Please update the digests file"
   rm $(DOWNLOAD_COOKIE)
# All looks good, proceed.
else
   cat $(DIGESTS_FILE) | grep $(LOCAL_FILE) | ( \
   ...
```